From 5a80a10a31553a1c46650996e79b7869076f92cc Mon Sep 17 00:00:00 2001
From: Stefan Kost <ensonic@users.sf.net>
Date: Tue, 21 Jul 2009 17:48:33 +0300
Subject: [PATCH] camerabin: small speedup in software zoom

Force crop on even pixel position to avoid slowpath in crop & scale.
---
 gst/camerabin/gstcamerabin.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/gst/camerabin/gstcamerabin.c b/gst/camerabin/gstcamerabin.c
index f119dea..feb15cb 100644
--- a/gst/camerabin/gstcamerabin.c
+++ b/gst/camerabin/gstcamerabin.c
@@ -1031,6 +1031,10 @@ gst_camerabin_set_element_zoom (GstCameraBin * camera, gint zoom)
     if (zoom != ZOOM_1X) {
       w2_crop = (camera->width - (camera->width * ZOOM_1X / zoom)) / 2;
       h2_crop = (camera->height - (camera->height * ZOOM_1X / zoom)) / 2;
+
+      /* force number of pixels cropped from left to be even, to avoid slow code
+       * path on videoscale */
+      w2_crop &= 0xFFFE;
     }
 
     pad_zoom_sink = gst_element_get_static_pad (camera->src_zoom_crop, "sink");
